### ISSUE TITLE
[Feature request] Set multiple fields at once

### ISSUE THREAD

**Issue by Vilin97** [JuliaObjects/Accessors.jl#119]
Setting several fields at once would reduce the number of times the immutable struct is copied. For example,
```
nt = (a=1, b=2)
@reset nt.a=2, nt.b=3
```
instead of 
```
nt = (a=1, b=2)
@reset nt.a=2
@reset nt.b=3
```

**Comment by Vilin97** [comment]
I am mostly looking for performance.
How can constructing a new struct be free? If a struct has `n` fields, how can creating a new struct not be `O(n)` time?

**Comment by ronisbr** [comment]
Yes! I can confirm @aplavin statement. In all cases I used Accessors.jl, I saw no performance penalty by setting multiple parameters at one time with `ConstructionBase.setproperties` ou using multiple `@reset`s. However, I think something like this would be less verbose:

```julia
@resets nt begin
    nt.a = 1
    nt.b = 2
    nt.c = 3
    ....
end
```

This idea is to replace all expressions with `nt.a` to `@reset nt.a` inside this block.
